src/MessagePack/MessagePackSerializer.NonGeneric.cs

```diff
@@ -7,13 +7,20 @@
 using System.Linq;
 using System.Linq.Expressions;
 using System.Reflection;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace MessagePack
 {
     public partial class MessagePackSerializer
     {
+#if NETFRAMEWORK || NETSTANDARD2_0
+        private static readonly bool AvoidDynamicCode = false;
+#else
+        private static readonly bool AvoidDynamicCode = !RuntimeFeature.IsDynamicCodeSupported;
+#endif
+
```

Comment:
Is it better to use the symbol `NETSTANDARD2_1`?

```csharp
#if NETSTANDARD2_1 || NETSTANDARD2_1_OR_GREATER
  private static readonly bool AvoidDynamicCode = !RuntimeFeature.IsDynamicCodeSupported;
#else
  private static readonly bool AvoidDynamicCode = false;
#endif
```

Related web page: [the description of the conditional compile of Unity 2022.3](https://docs.unity3d.com/2022.3/Documentation/Manual/PlatformDependentCompilation.html).

Reply:
It could go either way, perhaps. But using _only_ `NETSTANDARD2_1` will not apply to .NET 5+, and Unity will presumably someday move beyond .NET Standard 2.1. I'd rather specify the frameworks that don't and won't ever support the API so that I never have to modify this again.